source/common/stats/tag_utility.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "envoy/stats/symbol_table.h"
+#include "envoy/stats/tag.h"
+
+namespace Envoy {
+namespace Stats {
+namespace TagUtility {
+SymbolTable::StoragePtr addTagSuffix(StatName name, const StatNameTagVector& tags,
```

Comment:
Add doxygen doc. Also specify here that the returned StoragePtr is valid only through the lifetime of 'name' and 'tags'. That's because join() does not bump ref-counts in the symbol-table, to avoid taking a lock. But it is not obvious at the call-site that the returned StoragePtr has this property.

See the doc for SymbolTable::join() for details (ie just copy that as needed).

Also use doxygen format and consider the optional suggestion to make TagUtility an actual class